File: internal/templates/templates/webhook-relay/_node/services/process-webhook/handler.mjs
// webhooks queue — handle each delivery. Throwing = "retry me"; after 3
// attempts the message moves to webhooks-dlq.
export const handler = async (event) => {
  for (const record of event.Records) {
    const hook = JSON.parse(record.body);
    const attempt = Number(record.attributes.ApproximateReceiveCount);

    if (hook.fail) { // demo: {"fail": true} exercises retries + DLQ
      throw new Error(`webhook ${hook.id} failed on purpose (attempt ${attempt})`);
    }

    // Real work goes here: verify signature, update your systems, call APIs.
    console.log(`processed webhook ${hook.id} · type ${hook.type ?? "unknown"} (attempt ${attempt})`);
  }
};

File: internal/templates/templates/webhook-relay/_python/services/process-webhook/handler.py
"""webhooks queue — handle each delivery. Raising = "retry me"; after 3
attempts the message moves to webhooks-dlq."""

import json


def handler(event, context):
    for record in event["Records"]:
        hook = json.loads(record["body"])
        attempt = int(record["attributes"]["ApproximateReceiveCount"])

        if hook.get("fail"):  # demo: {"fail": true} exercises retries + DLQ
            raise RuntimeError(f"webhook {hook['id']} failed on purpose (attempt {attempt})")

        # Real work goes here: verify signature, update your systems, call APIs.
        print(f"processed webhook {hook['id']} · type {hook.get('type', 'unknown')} (attempt {attempt})")
